### PR TITLE
Heatmap layer

### PR DESCRIPTION
**Pull request by Nour-Cheour10**

### CODE REVIEW

**Comment by Nour-Cheour10** [comment]
update galata references

**Comment by Nour-Cheour10** [comment]
update galata references

**Comment by Nour-Cheour10** [comment]
update galata references

**Comment by Nour-Cheour10** [comment]
update galata references

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
   },
   "dependencies": {
     "@jupyter-widgets/base": "^1.1.10 || ^2 || ^3 || ^4 || ^5 || ^6",
+    "@jupyterlab/galata": "^5.2.2",
```
Comment (martinRenou) [inline package.json]:
```suggestion
```

I believe we don't need this

File: src/widget.ts [hunk 2]
```diff
@@ -69,12 +71,24 @@ export class MapModel extends DOMWidgetModel {
...
-    this.mapContainer = document.createElement('div');
-    this.mapContainer.style.height = '500px';
-    this.mapContainer.style.width = '100%';
-    this.el.appendChild(this.mapContainer);
+    this.el.classList.add('jupyter-widgets');
+    this.el.classList.add('ipyopenlayer-widgets');
+
+    this.map_container = document.createElement('div');
+    this.map_container.classList.add('ol-container');
+    requestAnimationFrame(() => {
```
Comment (martinRenou) [inline src/widget.ts:79]: It's unclear to me why we need a `requestAnimationFrame` here, what was the reason for adding it?

File: ui-tests/tests/ipyopenlayers.test.ts-snapshots/light-map-update-ipynb-cell-1-linux.png
Comment (martinRenou) [inline ui-tests/tests/ipyopenlayers.test.ts-snapshots/light-map-update-ipynb-cell-1-linux.png:1]: This is odd, the zoom control shows up but not the tile layer. I wonder if we should not sleep a tiny bit of time before doing the screenshot

File: src/widget.ts (hunk 2 above)
Comment (Nour-Cheour10) [reply src/widget.ts:79]: I was trying to get the parentelement with this : this.el.parentElement but it was returning null at first, So I added requestAnimationFrame to give it more time to get executed and to get the right answer.

File: ui-tests/tests/ipyopenlayers.test.ts-snapshots/light-map-update-ipynb-cell-1-linux.png
Comment (Nour-Cheour10) [reply ui-tests/tests/ipyopenlayers.test.ts-snapshots/light-map-update-ipynb-cell-1-linux.png:1]: Okay I'll add it

File: package.json (hunk 1 above)
Comment (Nour-Cheour10) [reply package.json]: We probably need it because when I committed it, it broked the CI test.

**Comment by Nour-Cheour10** [comment]
update galata references

File: ipyopenlayers/Map.py [hunk 3]
```diff
@@ -31,14 +31,35 @@ class TileLayer(Layer):
...
+    _model_name = Unicode('HeatmapLayerModel').tag(sync=True)
+    points= List([]).tag(sync=True)
+    blur =Int(15).tag(sync=True)
+    radius = Int(8).tag(sync=True)
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.points = kwargs.get('points', [])
+        self.blur = kwargs.get('blur', 15)
+        self.radius = kwargs.get('radius', 8)
```
Comment (martinRenou) [inline ipyopenlayers/Map.py]: This is redundant with the default traitlets values

```suggestion
```

File: ipyopenlayers/Map.py [hunk 4]
```diff
@@ -31,14 +31,35 @@ class TileLayer(Layer):
...
     _model_name = Unicode('OpenLayersGeoJSONModel').tag(sync=True)
     data = Dict({}).tag(sync=True)
     style = Dict({}).tag(sync=True)
     visible = Bool(True).tag(sync=True)
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.points = kwargs.get('points', [])
+        self.blur = kwargs.get('blur', 15)
+        self.radius = kwargs.get('radius', 8)
```
Comment (martinRenou) [inline ipyopenlayers/Map.py]: This seems to be unrelated to the GeoJSON class

```suggestion
```

**Review by martinRenou — CHANGES_REQUESTED** [review]

**Comment by martinRenou** [comment]
Thanks!
